File: Airship/AirshipCore/Source/Internal/UATaskRequest.m
/* Copyright Airship and Contributors */

#import "UATaskRequest+Internal.h"

@interface UATaskRequest()
@property(nonatomic, copy) NSString *taskID;
@property(nonatomic, strong) UATaskRequestOptions *options;
@property(nonatomic, strong) UATaskLauncher *launcher;
@end

@implementation UATaskRequest

- (instancetype)initWithID:(NSString *)taskID
                   options:(UATaskRequestOptions *)options
                  launcher:(UATaskLauncher *)launcher {
    self = [super init];
    if (self) {
        self.taskID = taskID;
        self.options = options;
        self.launcher = launcher;
    }
    return self;
}


+ (instancetype)requestWithID:(nonnull NSString *)taskID
                      options:(UATaskRequestOptions *)options
                     launcher:(nonnull UATaskLauncher *)launcher {

    return [[self alloc] initWithID:taskID
                            options:options
                           launcher:launcher];
}

@end

File: Airship/AirshipCore/Source/UATaskRequest.swift
/* Copyright Airship and Contributors */

/**
 * @note For internal use only. :nodoc:
 */
class UATaskRequest {
    let taskID: String
    let options: UATaskRequestOptions
    let launcher: UATaskLauncher

    init(taskID: String, options: UATaskRequestOptions, launcher: UATaskLauncher) {
        self.taskID = taskID
        self.options = options
        self.launcher = launcher
    }
}
